test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,318 @@
+	mutateSpecForAddingCustomDNS := func(pod *corev1.Pod) {
+		pod.Spec.DNSPolicy = corev1.DNSNone
+		if pod.Spec.DNSConfig == nil {
+			pod.Spec.DNSConfig = &corev1.PodDNSConfig{}
+		}
+		pod.Spec.DNSConfig.Nameservers = []string{dnsServiceIP}
+
+	}
+	//TODO: toolboxPodName ?
+	require.NoError(t, NewPodBuilder(toolboxPodName, data.testNamespace, ToolboxImage).
+		WithLabels(map[string]string{"app": "fqdn-cache-test"}).
+		WithContainerName(toolboxContainerName).
+		WithMutateFunc(mutateSpecForAddingCustomDNS).
+		Create(data))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, toolboxPodName, data.testNamespace))
+}
+
+// createHttpAgnHostPod creates an agnHost pod that serves HTTP requests and returns the IP of pod created.
+func createHttpAgnHostPod(t *testing.T, data *TestData) *PodIPs {
+	const (
+		agnHostPort          = 80
+		agnHostPodNamePreFix = "agnhost-"
+	)
+	podName := randName(agnHostPodNamePreFix)
+	args := []string{"netexec", "--http-port=" + strconv.Itoa(agnHostPort)}
+	ports := []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: agnHostPort,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+
+	require.NoError(t, NewPodBuilder(podName, data.testNamespace, agnhostImage).
+		WithArgs(args).
+		WithPorts(ports).
+		WithLabels(map[string]string{"app": "agnhost"}).
+		Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, podName, data.testNamespace)
+	require.NoError(t, err)
+	return podIPs
+}
+
+// createDnsPod creates the CoreDNS pod configured to use the custom DNS ConfigMap.
+func createCustomDnsPod(t *testing.T, data *TestData) {
+	volume := []corev1.Volume{
+		{
+			Name: "config-volume",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "custom-dns-config",
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "Corefile",
+							Path: "Corefile",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	volumeMount := []corev1.VolumeMount{
+		{
+			Name:      "config-volume",
+			MountPath: "/etc/coredns",
+		},
+	}
+
+	require.NoError(t, NewPodBuilder("custom-dns-server", data.testNamespace, "coredns/coredns:1.11.3").
+		WithLabels(map[string]string{"app": "custom-dns"}).
+		WithContainerName("coredns").
+		WithArgs([]string{"-conf", "/etc/coredns/Corefile"}).
+		AddVolume(volume).AddVolumeMount(volumeMount).
+		Create(data))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, "custom-dns-server", data.testNamespace))
+}
+
+// createDnsConfig generates a DNS configuration for the specified IP address and domain name.
+func createDnsConfig(t *testing.T, hosts map[string]string) map[string]string {
+	const coreFileTemplate = `lfx.test:53 {
+        errors
+        log
+        health
+        hosts {
+            {{ if .IPv4 }}{{ .IPv4 }} {{ $.FQDN }}{{ end }}
+            no_reverse
+            pods verified
+            ttl 10
```

Comment:
I think having the TTL has a function parameter would be a good idea
Could you explain why we are setting it to 10s and not 5s?

Reply:
Initially I had thought it to be a minimum enough value suitable to be used as TTL . Yes , using it as a parameter is a good idea, updated  the code for the same.